tokio/src/net/tcp/stream.rs

```diff
@@ -184,6 +184,56 @@ impl TcpStream {
         Ok(TcpStream { io })
     }
 
+    /// Turn a [`tokio::net::TcpStream`] into a [`std::net::TcpStream`].
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// use std::error::Error;
+    /// use std::io::Read;
+    /// use tokio::net::TcpListener;
+    /// # use tokio::net::TcpStream;
+    /// # use tokio::io::AsyncWriteExt;
+    /// # use tokio::time::Duration;
+    ///
+    /// #[tokio::main]
+    /// async fn main() -> Result<(), Box<dyn Error>> {
+    /// #   tokio::spawn(async {
+    /// #       tokio::time::sleep(Duration::from_millis(50)).await;
```

Comment:
I think you can avoid this sleep if you spawn it after calling `bind`. Also, can you await the returned `JoinHandle` at the end of the test to make sure the test fails if the task panics?